SigningInputEncoder.swift

```diff
@@ -0,0 +1,160 @@
+//Copyright DApps Platform Inc. All rights reserved.
+
+import Foundation
+import TrustWalletCore
+
+public struct SigningInputEncoder {
+    private init() { }
+    
+    public static func encode(data: Data, privateKey: Data, for coin: CoinType) throws -> SigningInput {
+        switch coin {
+        case .aeternity:
+            var input = try AeternitySigningInput(serializedData: data)
+            input.privateKey = privateKey
+            return input
+        case .aion:
+            var input = try AionSigningInput(serializedData: data)
+            input.privateKey = privateKey
+            return input
+        case .algorand:
+            var input = try AlgorandSigningInput(serializedData: data)
+            input.privateKey = privateKey
+            return input
+        case .binance:
+            var input = try BinanceSigningInput(serializedData: data)
+            input.privateKey = privateKey
+            return input
+        case .bitcoin,
+             .bitcoinCash,
+             .dash,
+             .digiByte,
+             .dogecoin,
+             .decred,
+             .groestlcoin,
+             .litecoin,
+             .monacoin,
+             .qtum,
+             .ravencoin,
+             .viacoin,
+             .zcash,
+             .zcoin,
+             .zelcash:
+            var input = try BitcoinSigningInput(serializedData: data)
+            input.privateKey = [ privateKey ]
```

Comment:
can we avoid supporting btc coins for now? as it requires multiple keys and etc, we can always add later